src/frontend-samples/explode-sample/ExplodeApp.tsx

```diff
@@ -0,0 +1,178 @@
+  public static explodeAttributes: ExplodeScalingAttributes = {
+    min: 1,
+    max: 3,
+    step: 0.05,
+  };
+  /** Uses the IModelTileRpcInterface API to query for the tile version info */
+  public static async queryTileFormatVersionInfo(): Promise<TileVersionInfo> {
+    return IModelTileRpcInterface.getClient().queryVersionInfo();
+  }
+  /** Uses the  EmphasizeElements API to isolate the elements related to the ids given. */
+  public static isolateElements(vp: Viewport, elementIds: string[]) {
+    const emph = EmphasizeElements.getOrCreate(vp);
+    emph.isolateElements(elementIds, vp);
+  }
+  /** Uses the  EmphasizeElements API to clear all isolated and emphasized. */
+  public static clearIsolateAndEmphasized(vp: Viewport) {
+    const emph = EmphasizeElements.getOrCreate(vp);
+    emph.clearIsolatedElements(vp);
+  }
+
+  /** Uses the IModel.js tools to fit the view to the object on screen. */
+  public static zoomToObject(vp: Viewport, objectName: string) {
+    const options: ViewChangeOptions = {
+      animateFrustumChange: true,
+      cancelOnAbort: false,
+    };
+
+    let volume = ExplodeTreeReference.getTreeRange(objectName);
+    if (undefined === volume) {
+      const awaitRangeLoaded: TreeDataListener = (name, rangeDidUpdate) => {
+        if (objectName === name && rangeDidUpdate) {
+          volume = ExplodeTreeReference.getTreeRange(objectName)!;
+          vp.zoomToVolume(volume, options);
+          ExplodeTreeReference.onTreeDataUpdated.removeListener(awaitRangeLoaded);
+        }
+      };
+      const removeListener = ExplodeTreeReference.onTreeDataUpdated.addListener(awaitRangeLoaded);
+      ExplodeApp.cleanUpCallbacks.push(removeListener);  // This will insure the listener is removed before swapping samples.
+    } else {
+      vp.zoomToVolume(volume, options);
+    }
+  }
+
+  /** Updates the tile tree reference with the given data and signals the viewport that changes have been made. */
+  public static refSetData(vp: Viewport, name: string, ids: string[], explodeScaling: number) {
+    const provider = ExplodeProvider.getOrCreate(vp);
+    provider.setData(name, ids, explodeScaling);
+
+    // Necessary when the data changes.
+    provider.invalidate();
+    // Necessary when the explode scaling changes.
+    ExplodeApp.invalidateScene(vp);
+  }
+
+  /** Uses the Viewport API to signal that the scene needs to re-render. */
+  public static invalidateScene(vp: Viewport) {
+    vp.invalidateScene();
+  }
+
+  /** Enables an animator using the Viewport API. If the animator is undefined, any active animator will be removed. */
+  public static setAnimator(vp: Viewport, animator?: Animator) {
+    vp.setAnimator(animator);
+  }
+}
+
+/** This provider both hides the original graphics of the element and inserts the transformed graphics. */
+class ExplodeProvider implements TiledGraphicsProvider, FeatureOverrideProvider {
+  // These methods support the provider as an object.
```

Comment:
// These methods support the provider as an object.

That comment seems entirely pointless.  Right?